storage/tests/SMAIndexSubBlock_unittest.cpp

```diff
@@ -0,0 +1,998 @@
+          .makeUncheckedComparatorForTypes(VarCharType::InstanceNonNullable(16), VarCharType::InstanceNonNullable(16)));
+
+  // Test with equals.
+  EXPECT_EQ(Selectivity::kNone,
+            sma_internal::getSelectivity(str_tvals[4], ComparisonID::kEqual,
+                                         str_tvals[0], str_tvals[3],
+                                         str_less.get(), str_equals.get()));
+  EXPECT_EQ(Selectivity::kNone,
+            sma_internal::getSelectivity(str_tvals[0], ComparisonID::kEqual,
+                                         str_tvals[1], str_tvals[3],
+                                         str_less.get(), str_equals.get()));
+  EXPECT_EQ(Selectivity::kSome,
+            sma_internal::getSelectivity(str_tvals[1], ComparisonID::kEqual,
+                                         str_tvals[0], str_tvals[2],
+                                         str_less.get(), str_equals.get()));
+}
+
+TEST_F(SMAIndexSubBlockTest, TestEvaluatePredicateCost) {
+  // Create the index on all the attributes.
+  createIndex({0, 1, 2, 3, 4, 5}, kIndexSubBlockSize);
+  const int min = 250, max = 500;
+  for (unsigned i = min; i <= max; ++i) {
+    if (i % 10 == 0) {
+      generateAndInsertTuple(i, true, "suffix");
+    } else {
+      generateAndInsertTuple(i, false, "suffix");
+    }
+  }
+  index_->rebuild();
+
+  // Test with inline values, returning both possible value types.
+  std::unique_ptr<ComparisonPredicate> predicate(
+      generateNumericComparisonPredicate<LongType>(ComparisonID::kLess, 0, min));
+  predicate_cost_t eval_cost = index_->estimatePredicateEvaluationCost(*predicate);
+  EXPECT_EQ(predicate_cost::kConstantTime, eval_cost);
+
+  predicate.reset(
+      generateNumericComparisonPredicate<LongType>(ComparisonID::kLess, 0, max + 1));
+  eval_cost = index_->estimatePredicateEvaluationCost(*predicate);
+  EXPECT_EQ(predicate_cost::kConstantTime, eval_cost);
+
+  predicate.reset(
+      generateNumericComparisonPredicate<LongType>(ComparisonID::kLess, 0, ((max - min)/2) + min));
+  eval_cost = index_->estimatePredicateEvaluationCost(*predicate);
+  EXPECT_EQ(predicate_cost::kInfinite, eval_cost);
+
+  // Test with out of line values.
+  predicate.reset(
+      generateStringComparisonPredicate(ComparisonID::kLess, 5, " "));
+  eval_cost = index_->estimatePredicateEvaluationCost(*predicate);
+  EXPECT_EQ(predicate_cost::kConstantTime, eval_cost);
+
+  predicate.reset(
+      generateStringComparisonPredicate(ComparisonID::kGreater, 5, " "));
+  eval_cost = index_->estimatePredicateEvaluationCost(*predicate);
+  EXPECT_EQ(predicate_cost::kConstantTime, eval_cost);
+
+  predicate.reset(
+      generateStringComparisonPredicate(ComparisonID::kEqual, 5, "300suffix"));
+  eval_cost = index_->estimatePredicateEvaluationCost(*predicate);
+  EXPECT_EQ(predicate_cost::kInfinite, eval_cost);
+
+  // Test with an inconsistent index.
+  tuple_id new_tuple = generateAndInsertTuple(0x1337, false, "suffix");
```

Comment:
After our discussion about supporting adHoc inserts, do you need to revisit this line? The index will be consistent after you insert the tuple, right?